### DISCUSSION TITLE
PostgreSQL issue - PANIC : replication checkpoint has wrong magic

### DISCUSSION THREAD

**Discussion by jun3280net** [teslamate-org/teslamate#1622] [Q&A] [answered]
Hi there,

I just tried reloading teslamate after noticing my VM environment had gone down.  However, now I'm receiving a POSTGRESQL error related to loading the database.   

database_1   | PostgreSQL Database directory appears to contain a database; Skipping initialization
database_1   | 
database_1   | 2021-05-14 08:54:33.423 UTC [1] LOG:  starting PostgreSQL 12.6 (Debian 12.6-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
database_1   | 2021-05-14 08:54:33.429 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
database_1   | 2021-05-14 08:54:33.433 UTC [1] LOG:  listening on IPv6 address "::", port 5432
database_1   | 2021-05-14 08:54:33.436 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
database_1   | 2021-05-14 08:54:33.453 UTC [26] LOG:  database system was interrupted; last known up at 2021-05-14 01:54:52 UTC
database_1   | 2021-05-14 08:54:33.477 UTC [26] PANIC:  replication checkpoint has wrong magic 1650864225 instead of 307747550
database_1   | 2021-05-14 08:54:34.455 UTC [28] FATAL:  the database system is starting up
database_1   | 2021-05-14 08:54:34.459 UTC [29] FATAL:  the database system is starting up
teslamate_1  | 2021-05-14 08:54:34.502 [error] Postgrex.Protocol (#PID<0.166.0>) failed to connect: ** (Postgrex.Error) FATAL 57P03 (cannot_connect_now) the database system is starting up
teslamate_1  | 2021-05-14 08:54:34.503 [error] Postgrex.Protocol (#PID<0.167.0>) failed to connect: ** (Postgrex.Error) FATAL 57P03 (cannot_connect_now) the database system is starting up
database_1   | 2021-05-14 08:54:34.952 UTC [1] LOG:  startup process (PID 26) was terminated by signal 6: Aborted
database_1   | 2021-05-14 08:54:34.952 UTC [1] LOG:  aborting startup due to startup process failure
database_1   | 2021-05-14 08:54:34.973 UTC [1] LOG:  database system is shut down

I've captured the log images below:

![image](https://user-images.githubusercontent.com/30555695/118250747-02b3d000-b464-11eb-8930-ee7d84e4edfd.png)
![image](https://user-images.githubusercontent.com/30555695/118251934-6f7b9a00-b465-11eb-8bd0-a469caa4823a.png)
![image](https://user-images.githubusercontent.com/30555695/118252217-c1bcbb00-b465-11eb-8b1a-06ec9ba7b5bc.png)

This is definitely outside the realm of my understanding, but I figured I might be able to backup the database and restore it.  However, backup is not possible as it complains the container is restarting : 
![image](https://user-images.githubusercontent.com/30555695/118253265-f2e9bb00-b466-11eb-80da-330c5c148b4f.png)

Any suggestions on how to troubleshoot this issue would be greatly appreciated.

Thank you

**Comment by jun3280net** [comment, marked as answer]
Hopefully, this helps other people out who run into this situation.  I found the following link which provides some insight into how to address the problem : 

[https://www.postgresql-archive.org/PANIC-replication-checkpoint-has-wrong-magic-0-instead-of-td6146772.html](https://www.postgresql-archive.org/PANIC-replication-checkpoint-has-wrong-magic-0-instead-of-td6146772.html)

My steps are as follows : 

- **docker-compose down**  or **CTRL + C** to shut down the Teslamate, Postgres, Grafana
- With "elevated privileges / root" I navigated to the directory in Linux Mint : **/var/lib/docker/volumes/teslamate_teslamate-db/_data/pg_logical/**
- Then I renamed the **replorigin_checkpoint** file to **replorigin_checkpoint_corrupted**, as starting up the docker-compose will generate a new **replorigin_checkpoint** file.  In theory, you could delete the file for the same effect - but renaming the file gives you the ability to revert back to your previous state if this doesn't work for you.

(this screenshot includes the newly generated replorigin_checkpoint file)
![image](https://user-images.githubusercontent.com/30555695/118303936-3eba5580-b4a3-11eb-8603-f69c6b6353ec.png)

- **docker-compose up** 
- All fixed :)

![image](https://user-images.githubusercontent.com/30555695/118304269-b25c6280-b4a3-11eb-9966-39aacf48c740.png)
![image](https://user-images.githubusercontent.com/30555695/118304366-d324b800-b4a3-11eb-97dd-4e9f43c4d8fa.png)

**Reply by an-suz** [reply]
Great, thanks for this. For anyone trying to fix the problem on MacOS (as I was), there's an extra step to find the Docker volume: https://ekartco.com/2019/12/enter-docker-vm-on-macos-catalina-ssh-xhyve/

**Reply by BlackWolfWoof** [reply]
You saved my butt, thanks a lot 🖤
Was using postgres for [Immich](https://immich.app/)

**Reply by mohammadrz003** [reply]
worked like a charm!
thanks man.

**Reply by cotunjr** [reply]
after 4 years, thysm. Fixed it with a just simple change.

**Reply by mrpond** [reply]
life saver!

![image](https://github.com/user-attachments/assets/7d3042ad-6c32-4086-973f-1df8e0637486)
